### PR TITLE
ICU-20121 Implementing FormattedList.

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20121
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [x] Tests included
- [x] Documentation is changed or added

### CODE REVIEW

**Comment by sffc** [comment]
@aheninger Let's go ahead and review this.  I anticipate adding Java, and perhaps making some minor changes here, but we can get this checked in first.

**Comment by sffc** [comment]
I put the "incomplete" tag back on because there are a couple other small things I need to do on this PR.  Please look at the FormattedDateInterval PR first (#406).

File: icu4c/source/i18n/unicode/ulistformatter.h [hunk 1]
```diff
@@ -145,6 +202,40 @@ ulistfmt_format(const UListFormatter* listfmt,
...
+ *            the number of entries in strings[], and the number of entries
+ *            in the stringLengths array if it is not NULL. Must be >= 0.
+ * @param uresult
+ *            The object in which to store the result of the list formatting
+ *            operation. See ulistfmt_openResult().
+ * @param status
+ *            Error code set if an error occurred during formatting.
+ */
+U_CAPI void U_EXPORT2
+ulistfmt_formatStringsToValue(
```
Comment (aheninger) [inline icu4c/source/i18n/unicode/ulistformatter.h]: Does this want to change to formatStringsToResult(), to be consistent with the UFormattedValue Function Name Conventions discussions?

Comment (sffc) [reply icu4c/source/i18n/unicode/ulistformatter.h]: Done.

**Pull request merged**